src/legacy/server/saved_objects/lib/cache_index_patterns.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { FieldDescriptor } from 'src/legacy/server/index_patterns/service/index_patterns_service';
+import { IndexPatternsService } from 'src/legacy/server/index_patterns';
+import { SavedObjectsIndexPattern } from 'src/core/server';
+
+export class CacheIndexPatterns {
```

Comment:
Why does this live in the legacy platform? Seems like it should be in the `src/core/server/saved_objects/lib/` directory?